src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,235 @@
+package duke.parser;
+
+import duke.commands.Command;
+import duke.commands.DeadlineCommand;
+import duke.commands.DeleteCommand;
+import duke.commands.DoneCommand;
+import duke.commands.EventCommand;
+import duke.commands.ExitCommand;
+import duke.commands.FindCommand;
+import duke.commands.IncorrectCommand;
+import duke.commands.ListCommand;
+import duke.commands.TodoCommand;
+
+import duke.data.tasks.Deadline;
+import duke.data.tasks.Event;
+import duke.data.tasks.Todo;
+
+import java.util.regex.Pattern;
+import java.text.ParseException;
+
+/**
+ * Implements the Parser class which parses user input and returns commands to execute.
+ * @author Lim Yong Shen, Kevin
+ */
+public class Parser {
+
+    private static final String BASIC_COMMAND_FORMAT = "[\\w\\/\\:\\-\\s]+";
+    private static final int COMMAND_WORD_INDEX = 0;
+    private static final int ARGUMENTS_INDEX = 1;
+    private static final int INPUT_STRING_SPLIT_LENGTH_LIMIT = 2;
+
+    /**
+     * Parses the specified input and returns the appropriate command.
+     * @param input The specified input.
+     * @return The appropriate command based on the specified input.
+     */
+    public static Command parse(String input) {
+        if (!Pattern.matches(BASIC_COMMAND_FORMAT, input)) {
+            return prepareIncorrectCommand();
+        }
+        String commandWord = getCommandWord(input);
+        switch (commandWord) {
+        case DeadlineCommand.COMMAND_WORD:
+            return prepareDeadlineCommand(input);
+        case DeleteCommand.COMMAND_WORD:
+            return prepareDeleteCommand(input);
+        case DoneCommand.COMMAND_WORD:
+            return prepareDoneCommand(input);
+        case EventCommand.COMMAND_WORD:
+            return prepareEventCommand(input);
+        case ExitCommand.COMMAND_WORD:
+            return prepareExitCommand();
+        case FindCommand.COMMAND_WORD:
+            return prepareFindCommand(input);
+        case ListCommand.COMMAND_WORD:
+            return prepareListCommand();
+        case TodoCommand.COMMAND_WORD:
+            return prepareTodoCommand(input);
+        default:
```

Comment:
Maybe try using `enum`.

Reply:
Good point! I was thinking of using enums for other purposes but I chose not to do so here since the criteria for comparison is the first word of the input string. I did actually use an enum in an older version of the code which ultimately got refactored away. I'll keep this in mind for the GUI portion of the project!